server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -445,8 +445,19 @@ public Analyzer indexAnalyzer() {
         return this.indexAnalyzer;
     }
 
+    public NamedAnalyzer indexAnalyzer(String field, Function<String, NamedAnalyzer> unmappedFieldAnalyzer) {
```

Comment:
Maybe rename `unmappedFieldAnalyzer` into `defaultAnalyzer` ? That feels wrong to rely on something called unmapped for runtime fields ?

Reply:
Can you also add javadoc for the function ?